lua/plenary/path.lua

```diff
@@ -202,20 +202,31 @@ function Path:mkdir(opts)
   opts = opts or {}
 
   local mode = opts.mode or 448 -- 0700 -> decimal
-  local parents = F.if_nil(opts.parents, false)
-  local exists_ok = F.if_nil(opts.exists_ok, true)
+  local parents = F.if_nil(opts.parents, false, opts.parents)
+  local exists_ok = F.if_nil(opts.exists_ok, true, opts.exists_ok)
 
   if not exists_ok and self:exists() then
     error("FileExistsError:" .. self:absolute())
   end
 
   if not uv.fs_mkdir(self:absolute() or self.filename,  mode) then
     if parents then
-      -- TODO: Find all the parents
-      error("Not implemented")
+      local dirs = vim.split(self:absolute(), self._sep)
+      local processed = ''
+      for _, dir in ipairs(dirs) do
+        if dir ~= '' then
+          local joined = processed .. self._sep .. dir
```

Comment:
I think in general we shouldn't use string concat outside of small helper functions... too easy to mess up. We should wrap using seps with other stuff.